pkg/integration/tests/branch/new_branch_from_remote_tracking_same_name.go

```diff
@@ -0,0 +1,60 @@
+package branch
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var NewBranchFromRemoteTrackingSameName = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Set tracking information when creating a new branch from a remote branch",
+	ExtraCmdArgs: []string{},
+	Skip:         false,
+	SetupConfig:  func(config *config.AppConfig) {},
+	SetupRepo: func(shell *Shell) {
+		shell.NewBranch("main")
+		shell.EmptyCommitWithDate("commit", "2023-04-07 10:00:00")
+		shell.NewBranchFrom("other_branch", "main")
+		shell.CloneIntoRemote("origin")
+		shell.Checkout("main")
+		shell.RunCommand([]string{"git", "branch", "-D", "other_branch"})
+	},
+	Run: func(t *TestDriver, keys config.KeybindingConfig) {
+		t.Views().Remotes().
+			Focus().
+			Lines(
+				Contains("origin").IsSelected(),
+			).
+			PressEnter()
+
+		t.Views().RemoteBranches().
+			IsFocused().
+			Lines(
+				Contains("main").IsSelected(),
+				Contains("other_branch"),
+			).
+			SelectNextItem().
+			Press(keys.Universal.New)
+
+		t.ExpectPopup().Prompt().
+			Title(Equals("New branch name (branch is off of 'origin/other_branch')")).
+			Confirm()
+
+		t.Views().Branches().
+			Focus().
+			Lines(
+				Contains("other_branch").IsSelected(),
+				Contains("main"),
+			).
+			Press(keys.Branches.SetUpstream)
+
+		t.ExpectPopup().Menu().Title(Contains("Upstream")).Select(Contains("View divergence from upstream")).Confirm()
+
+		t.Views().SubCommits().
+			IsFocused().
+			Title(Contains("Commits (other_branch <-> origin/other_branch)")).
+			Lines(
+				DoesNotContainAnyOf("↓", "↑").Contains("--- Remote ---"),
+				DoesNotContainAnyOf("↓", "↑").Contains("--- Local ---"),
+			)
```

Comment:
Instead of doing all this, you could simply test that "other_branch" has a checkmark in the branches list (and doesn't in the other test). That's a bit less work, but maybe not quite as explicit. Unsure, just a thought.

Reply:
I don't mind changing it, I like that it is more explicit but I can see at least two good reasons to change it.
1. Less code can be easier to read, understand and modify
2. Less code can be faster to run

I think having a tight feedback loop on tests is important since they should be run often.

Reply:
So, let me know what you prefer, please.

Reply:
I agree that less code is better, so I prefer to change it to the shorter version.

Reply:
It's done.